testctrl/docs/specs/container_image_names.md

```diff
@@ -0,0 +1,119 @@
+
+Docker images are encapsulated in repositories in a registry. The naming is a
+bit confusing. In terms of the Docker CLI, a tag is `"httpd:v1"` where `"v1"`
+is the version. This differs from the Registry perspective, where `"httpd"` is
+a repository and `"v1"` is a tag.
+
+Docker registries, like Google Container Registry ([gcr.io]) and DockerHub
+([hub.docker.com]), implement a common [Registry API]. However, docker and this
+API do not natively provide batch clean-up actions. Docker also provides no TTL
+on container images.
+
+[gcr.io]: https://gcr.io
+[hub.docker.com]: https://hub.docker.com
+[Registry API]: https://docs.docker.com/registry/spec/api/
+
+One approach for building a TTL on container images would be to use timestamps
+as versions/tags of the images. However, the Registry API cannot query based on
+these. Checking these requires a separate API call for each repository.
+
+Instead of excessive requests, this framework imposes a strict set of naming
+conventions that can aid in automated clean up.
+
+## Specification
+
+The framework allows a repository to contain exactly 1 container image. The
+repository is named as follows:
+
+```
+[ REGISTRY_PREFIX "/" ]
+    "benchmarks" "/"
+          <USER> "/"
+              <TIMESTAMP> "/"
+                  <COMPONENT_KIND> "/"
+                      <LANGUAGE_CODE> [:latest]
+```
+
+1.  The name begins with an optional `REGISTRY_PREFIX`. Some registries, like
+    Google Container Registry (GCR) require a specific prefix. This allows
+    images to be uploaded to these registries.
+
+2.  The word `"benchmarks"` helps organize images. Projects may have many
+    images, and this keyword shows which are related to performance tests. In
+    GCR, these repositories are displayed in a nested form, like a subdirectory.
+
+3.  Each container contains a component's executable that was built after
+    cloning a GitHub repository. The `USER` specifies the username or
+    organization of this repository. For example, the grpc/grpc-java repository
+    would set the user to "grpc". A developer's fork, like codeblooded/grpc,
+    would set the user to "codeblooded".
+
+4.  The `TIMESTAMP` is the time when the set of images were built, represented in
+    seconds since the unix epoch.
+
+5.  The `COMPONENT_KIND` specifies what the image contains. For example, this
+    may be `"driver"` for the test orchestrator and `"worker"` for a client or
+    server.
+
+6.  The `LANGUAGE_CODE` indicates the programming language of the component.
+    This differentiates two components that may be different but built
+    simultaneously. It is omitted for component types which only have 1
+    language, like the `"driver"`. An example would be a load test which uses
+    both a worker in Go and a worker in C++.
+
+7.  The only relevant version/tag is `"latest"`, because only one image lives in
+    a repository.
+
+The user and component kind allow developers to quickly see who built the image
```

Comment:
> different TTLs for CI images vs user images might be useful, but the "user" as defined isn't probably the best criterion to distinguish between those.

I agree that this is probably not as useful for indicating who built the test. The "user" was more of a specification of the owner of the GitHub fork or the "grpc" organization itself. 

@gnossen made a request about not relying on git repositories (which will change the scheme). Let's push this discussion to our internal guide, since @gnossen's argument is there.